Duplicati/Library/Utility/FilterExpression.cs

```diff
@@ -186,9 +186,9 @@ private static Regex GetFilterGroupRegex(string filterGroupName)
             /// <note>From: http://www.c-sharpcorner.com/uploadfile/b81385/efficient-string-matching-algorithm-with-use-of-wildcard-characters/</note>
             private static bool IsWildcardMatch(string input, string pattern)
             {
-                int[] inputPosStack = new int[(input.Length + 1) * (pattern.Length + 1)];   // Stack containing input positions that should be tested for further matching
-                int[] patternPosStack = new int[inputPosStack.Length];                      // Stack containing pattern positions that should be tested for further matching
-                int stackPos = -1;                                                          // Points to last occupied entry in stack; -1 indicates that stack is empty
+                Stack<int> inputPosStack = new Stack<int>(); // Stack containing input positions that should be tested for further matching
+                Stack<int> patternPosStack = new Stack<int>(); // Stack containing pattern positions that should be tested for further matching
+
                 bool[,] pointTested = new bool[input.Length + 1, pattern.Length + 1];       // Each true value indicates that input position vs. pattern position has been tested

```

Comment:
I made several attempts to replace this multidimensional array with various data structures (`HashSet<Tuple<int, int>>`, `bool[][]`), but all my attempts proved unsuccessful in improving the performance.  For some reason, this allocation does not seem to have as large of an effect on performance as the removed array allocations above.